src/CppParser/Parser.cpp

```diff
@@ -887,9 +887,11 @@ static bool HasLayout(const clang::RecordDecl* Record)
 
 bool Parser::IsSupported(const clang::NamedDecl* ND)
 {
-    return !c->getSourceManager().isInSystemHeader(ND->getLocStart()) ||
-        (llvm::isa<clang::RecordDecl>(ND) &&
-         supportedStdTypes.find(ND->getName()) != supportedStdTypes.end());
+
+	return !ND->getSourceRange().isValid() ||
+		!c->getSourceManager().isInSystemHeader(ND->getLocStart()) ||
```

Comment:
Please keep the alignment we had before - just 4 spaces per line. Basically, the code should be aligned the same way as before, with the exception of the newly added first line.